pkg/validation/cluster.go

```diff
@@ -254,7 +275,7 @@ func ValidateUpdateCluster(ctx context.Context, newCluster, oldCluster *kubermat
 	}
 
 	secretKeySelectorFunc := provider.SecretKeySelectorValueFuncFactory(ctx, clusterProvider.GetSeedClusterAdminRuntimeClient())
-	cloudProvider, err := cloud.Provider(dc, secretKeySelectorFunc, caBundle)
+	cloudProvider, err := cloud.Provider(dc, secretKeySelectorFunc, caBundle, resources.DefaultNodePortRange)
```

Comment:
Shouldn't we pass the "real" range here, i.e. the range provided to the seed-controller-manager? Passing the default range seems a little bit strange to me. If we would ever use that value in the context of validation, we would always validate the default value, which might not be used at all.

On that note, are we validating that value anywhere?

Reply:
No, it's OK here. It doesn't affect anything actually and an empty string could be passed as well.

NodePortRange parameter (just like caBundle) is a leaking API here, that indicates the need to refactor the cloud provider interface, but it's out of scope of this PR. And I've experienced that need first-hand.

The seed-manager flag `-node-port-range` is inaccessible in this context, as this function is part of web API, without deep refactoring it does not seem possible.